libs/designer-ui/src/lib/unitTesting/outputMocks/errorDetails.tsx

```diff
@@ -0,0 +1,42 @@
+import { useCallback } from 'react';
+import { TextField, type ITextFieldProps } from '@fluentui/react';
+import type { ChangeState } from 'lib/editor/base';
+
+export interface ErrorDetailsProps {
+  errorMessage: string;
+  onErrorMessageChange: (value: string) => void;
+  errorCode: string;
+  onErrorCodeChange: (value: string) => void;
+  onMockUpdate: (id: string, newState: ChangeState, type: string) => void; // Added prop
+}
+
+const ErrorDetails: React.FC<ErrorDetailsProps> = ({ errorMessage, onErrorMessageChange, errorCode, onErrorCodeChange, onMockUpdate }) => {
+  const handleErrorMessageChange: ITextFieldProps['onChange'] = useCallback(
+    (_: any, newValue: string | undefined) => {
+      if (newValue !== undefined) {
+        onErrorMessageChange(newValue);
+        onMockUpdate('errorMessage', { value: newValue } as unknown as ChangeState, 'string'); // Trigger mock update
+      }
+    },
+    [onErrorMessageChange, onMockUpdate]
+  );
+
+  const handleErrorCodeChange: ITextFieldProps['onChange'] = useCallback(
+    (_: any, newValue: string | undefined) => {
+      if (newValue !== undefined) {
+        onErrorCodeChange(newValue);
+        onMockUpdate('errorCode', { value: newValue } as unknown as ChangeState, 'string'); // Trigger mock update
+      }
+    },
+    [onErrorCodeChange, onMockUpdate]
+  );
+
+  return (
+    <div className="error-fields">
```

Comment:
NIT: are you using this classname? or is this classname setting some styles?

Reply:
Thanks carlos - FYI Changes made:

Removed the "error-fields" className from the div.
Added a basic inline style to the div for better default appearance.
Updated the type for the first parameter of the onChange handlers to be more specific.
Removed the explicit type annotation for handleErrorMessageChange and handleErrorCodeChange, letting TypeScript infer the types.